lib/daemon.py

```diff
@@ -145,7 +155,18 @@ def check_sh_is_running(pidfile):
     """
     
     pid = read_pidfile(pidfile)
-    return psutil.pid_exists(pid) if pid > 0 else False
+    isRunning = False
+    if pid > 0 and psutil.pid_exists(pid):
+        try:
+            fd = os.open(pidfile, os.O_RDONLY)
+            fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+            os.close(fd)
+            # pidfile not locked, so sh is terminated
+        except OSError as e:
+            # pidfile is locked, so sh is running
+            isRunning = True
+            pass
```

Comment:
Do we need a `pass` here?

Further if locking the file fails, the file is still opened. Maybe the `os.close(fd)` should be in the `except` block.

Reply:
moved to finally block